from PyQt5 import QtCore, QtGui, QtWidgets
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'Backend/test'))
import scapyUtilities, scapySniffer

class Ui_PacketFromPCAPView(object):
    def setupUi(self, Form):
        Form.setObjectName("Form")
        Form.resize(924, 884)
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout(Form)
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.gridLayout = QtWidgets.QGridLayout()
        self.gridLayout.setObjectName("gridLayout")
        self.PcapFile = QtWidgets.QGroupBox(Form)
        self.PcapFile.setMaximumSize(QtCore.QSize(16777215, 80))
        self.PcapFile.setObjectName("PcapFile")
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout(self.PcapFile)
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.PCAFile_2 = QtWidgets.QLabel(self.PcapFile)
        self.PCAFile_2.setObjectName("PCAFile_2")
        self.horizontalLayout_3.addWidget(self.PCAFile_2)
        self.PCAPLocation_2 = QtWidgets.QPlainTextEdit(self.PcapFile)
        self.PCAPLocation_2.setInputMethodHints(QtCore.Qt.ImhNone)
        self.PCAPLocation_2.setVerticalScrollBarPolicy(QtCore.Qt.ScrollBarAsNeeded)
        self.PCAPLocation_2.setObjectName("PCAPLocation_2")
        self.horizontalLayout_3.addWidget(self.PCAPLocation_2)
        self.gridLayout_2 = QtWidgets.QGridLayout()
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.horizontalLayout_3.addLayout(self.gridLayout_2)
        self.OpenButton_2 = QtWidgets.QPushButton(self.PcapFile)
        self.OpenButton_2.setObjectName("OpenButton_2")
        self.OpenButton_2.clicked.connect(self.openPCAP)
        self.horizontalLayout_3.addWidget(self.OpenButton_2)
        self.CancelButton_2 = QtWidgets.QPushButton(self.PcapFile)
        self.CancelButton_2.setObjectName("CancelButton_2")
        self.horizontalLayout_3.addWidget(self.CancelButton_2)
        self.gridLayout.addWidget(self.PcapFile, 0, 0, 1, 1)
        self.PacketArea = QtWidgets.QGroupBox(Form)
        self.PacketArea.setObjectName("PacketArea")
        self.gridLayout_7 = QtWidgets.QGridLayout(self.PacketArea)
        self.gridLayout_7.setObjectName("gridLayout_7")
        self.ClearButton_4 = QtWidgets.QPushButton(self.PacketArea)
        self.ClearButton_4.setObjectName("ClearButton_4")
        self.gridLayout_7.addWidget(self.ClearButton_4, 0, 1, 1, 1)
        self.gridLayout_6 = QtWidgets.QGridLayout()
        self.gridLayout_6.setObjectName("gridLayout_6")
        self.gridLayout_7.addLayout(self.gridLayout_6, 0, 2, 1, 1)
        self.PacketView_2 = QtWidgets.QTabWidget(self.PacketArea)
        self.PacketView_2.setObjectName("PacketView_2")
        self.Dissected_2 = QtWidgets.QWidget(self.PacketView_2) #linked to packet area?
        self.Dissected_2.setObjectName("Dissected_2")
        self.treeWidget = QtWidgets.QTreeWidget(self.Dissected_2)
        self.treeWidget.setGeometry(QtCore.QRect(0, 0, 781, 291))
        self.treeWidget.setObjectName("treeWidget")
        item_0 = QtWidgets.QTreeWidgetItem(self.treeWidget)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        item_0 = QtWidgets.QTreeWidgetItem(self.treeWidget)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
<<<<<<< HEAD
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        for i in range(2):
            item_0 = self.addDropDownMumboJumbo(self.treeWidget)
            item_1 = self.addDropDownMumboJumbo(item_0)
            item_1 = self.addDropDownMumboJumbo(item_0)

=======
>>>>>>> 4bc1659acff2391f05aad74e7d466a2641c221f2
        self.PacketView_2.addTab(self.Dissected_2, "")
        self.Binary_2 = QtWidgets.QWidget()
        self.Binary_2.setObjectName("Binary_2")
        self.treeWidget_2 = QtWidgets.QTreeWidget(self.Binary_2)
        self.treeWidget_2.setGeometry(QtCore.QRect(0, 0, 781, 291))
        self.treeWidget_2.setObjectName("treeWidget_2")
        item_0 = QtWidgets.QTreeWidgetItem(self.treeWidget_2)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        item_0 = QtWidgets.QTreeWidgetItem(self.treeWidget_2)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        self.PacketView_2.addTab(self.Binary_2, "")
        self.HEX_2 = QtWidgets.QWidget()
        self.HEX_2.setObjectName("HEX_2")
        self.treeWidget_3 = QtWidgets.QTreeWidget(self.HEX_2)
        self.treeWidget_3.setGeometry(QtCore.QRect(0, 0, 781, 291))
        self.treeWidget_3.setObjectName("treeWidget_3")
        item_0 = QtWidgets.QTreeWidgetItem(self.treeWidget_3)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        item_0 = QtWidgets.QTreeWidgetItem(self.treeWidget_3)
        item_1 = QtWidgets.QTreeWidgetItem(item_0)
        self.PacketView_2.addTab(self.HEX_2, "")
        self.gridLayout_7.addWidget(self.PacketView_2, 0, 0, 1, 1)
        self.gridLayout.addWidget(self.PacketArea, 2, 0, 1, 1)
        self.CapturFilterArea = QtWidgets.QGroupBox(Form)
        self.CapturFilterArea.setMaximumSize(QtCore.QSize(16777215, 80))
        self.CapturFilterArea.setObjectName("CapturFilterArea")
        self.gridLayout_5 = QtWidgets.QGridLayout(self.CapturFilterArea)
        self.gridLayout_5.setObjectName("gridLayout_5")
        self.FilterLabel_2 = QtWidgets.QLabel(self.CapturFilterArea)
        self.FilterLabel_2.setObjectName("FilterLabel_2")
        self.gridLayout_5.addWidget(self.FilterLabel_2, 0, 0, 1, 1)
        self.FilterExpressionInput_2 = QtWidgets.QPlainTextEdit(self.CapturFilterArea)
        self.FilterExpressionInput_2.setInputMethodHints(QtCore.Qt.ImhNone)
        self.FilterExpressionInput_2.setVerticalScrollBarPolicy(QtCore.Qt.ScrollBarAsNeeded)
        self.FilterExpressionInput_2.setObjectName("FilterExpressionInput_2")
        self.gridLayout_5.addWidget(self.FilterExpressionInput_2, 0, 1, 1, 1)
        self.ApplyButton_2 = QtWidgets.QPushButton(self.CapturFilterArea)
        self.ApplyButton_2.setObjectName("ApplyButton_2")
        self.gridLayout_5.addWidget(self.ApplyButton_2, 0, 2, 1, 1)
        self.ClearButton_3 = QtWidgets.QPushButton(self.CapturFilterArea)
        self.ClearButton_3.setObjectName("ClearButton_3")
        self.gridLayout_5.addWidget(self.ClearButton_3, 0, 3, 1, 1)
        self.gridLayout_4 = QtWidgets.QGridLayout()
        self.gridLayout_4.setObjectName("gridLayout_4")
        self.gridLayout_5.addLayout(self.gridLayout_4, 1, 2, 1, 2)
        self.gridLayout.addWidget(self.CapturFilterArea, 1, 0, 1, 1)
        self.FieldandFuzzing = QtWidgets.QGroupBox(Form)
        self.FieldandFuzzing.setTitle("")
        self.FieldandFuzzing.setObjectName("FieldandFuzzing")
        self.gridLayout_9 = QtWidgets.QGridLayout(self.FieldandFuzzing)
        self.gridLayout_9.setObjectName("gridLayout_9")
        self.groupBox = QtWidgets.QGroupBox(self.FieldandFuzzing)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.groupBox.sizePolicy().hasHeightForWidth())
        self.groupBox.setSizePolicy(sizePolicy)
        self.groupBox.setMinimumSize(QtCore.QSize(0, 150))
        self.groupBox.setSizeIncrement(QtCore.QSize(0, 0))
        self.groupBox.setObjectName("groupBox")
        self.gridLayout_8 = QtWidgets.QGridLayout(self.groupBox)
        self.gridLayout_8.setObjectName("gridLayout_8")
        self.Forward_2 = QtWidgets.QPushButton(self.groupBox)
        self.Forward_2.setObjectName("Forward_2")
        self.gridLayout_8.addWidget(self.Forward_2, 9, 2, 1, 1)
        self.SaveMod_2 = QtWidgets.QPushButton(self.groupBox)
        self.SaveMod_2.setObjectName("SaveMod_2")
        self.gridLayout_8.addWidget(self.SaveMod_2, 9, 1, 1, 1)
        self.Drop_2 = QtWidgets.QPushButton(self.groupBox)
        self.Drop_2.setObjectName("Drop_2")
        self.gridLayout_8.addWidget(self.Drop_2, 9, 3, 1, 1)
        self.checkBox_10 = QtWidgets.QCheckBox(self.groupBox)
        self.checkBox_10.setText("")
        self.checkBox_10.setObjectName("checkBox_10")
        self.gridLayout_8.addWidget(self.checkBox_10, 3, 0, 1, 1)
        self.comboBox_8 = QtWidgets.QComboBox(self.groupBox)
        self.comboBox_8.setObjectName("comboBox_8")
        self.comboBox_8.addItem("")
        self.comboBox_8.addItem("")
        self.comboBox_8.addItem("")
        self.comboBox_8.addItem("")
        self.gridLayout_8.addWidget(self.comboBox_8, 1, 4, 1, 1)
        self.checkBox_9 = QtWidgets.QCheckBox(self.groupBox)
        self.checkBox_9.setText("")
        self.checkBox_9.setObjectName("checkBox_9")
        self.gridLayout_8.addWidget(self.checkBox_9, 4, 0, 1, 1)
        self.FieldValuesCol_2 = QtWidgets.QTreeWidget(self.groupBox)
        self.FieldValuesCol_2.setMinimumSize(QtCore.QSize(411, 192))
        self.FieldValuesCol_2.setSizeIncrement(QtCore.QSize(0, 0))
        self.FieldValuesCol_2.setIndentation(10)
        self.FieldValuesCol_2.setUniformRowHeights(True)
        self.FieldValuesCol_2.setItemsExpandable(True)
        self.FieldValuesCol_2.setObjectName("FieldValuesCol_2")
        item_0 = QtWidgets.QTreeWidgetItem(self.FieldValuesCol_2)
        font = QtGui.QFont()
        font.setPointSize(10)
        item_0.setFont(0, font)
        item_0 = QtWidgets.QTreeWidgetItem(self.FieldValuesCol_2)
        item_0 = QtWidgets.QTreeWidgetItem(self.FieldValuesCol_2)
        item_0 = QtWidgets.QTreeWidgetItem(self.FieldValuesCol_2)
        item_0 = QtWidgets.QTreeWidgetItem(self.FieldValuesCol_2)
        self.gridLayout_8.addWidget(self.FieldValuesCol_2, 0, 1, 8, 3)
        self.label_2 = QtWidgets.QLabel(self.groupBox)
        self.label_2.setObjectName("label_2")
        self.gridLayout_8.addWidget(self.label_2, 8, 1, 1, 2)
        self.checkBox_6 = QtWidgets.QCheckBox(self.groupBox)
        self.checkBox_6.setSizeIncrement(QtCore.QSize(0, 2))
        self.checkBox_6.setText("")
        self.checkBox_6.setObjectName("checkBox_6")
        self.gridLayout_8.addWidget(self.checkBox_6, 5, 0, 1, 1)
        self.checkBox_7 = QtWidgets.QCheckBox(self.groupBox)
        self.checkBox_7.setText("")
        self.checkBox_7.setObjectName("checkBox_7")
        self.gridLayout_8.addWidget(self.checkBox_7, 1, 0, 1, 1)
        self.checkBox_8 = QtWidgets.QCheckBox(self.groupBox)
        self.checkBox_8.setText("")
        self.checkBox_8.setObjectName("checkBox_8")
        self.gridLayout_8.addWidget(self.checkBox_8, 2, 0, 1, 1)
        self.comboBox_10 = QtWidgets.QComboBox(self.groupBox)
        self.comboBox_10.setObjectName("comboBox_10")
        self.comboBox_10.addItem("")
        self.comboBox_10.addItem("")
        self.comboBox_10.addItem("")
        self.comboBox_10.addItem("")
        self.gridLayout_8.addWidget(self.comboBox_10, 2, 4, 1, 1)
        self.comboBox_9 = QtWidgets.QComboBox(self.groupBox)
        self.comboBox_9.setObjectName("comboBox_9")
        self.comboBox_9.addItem("")
        self.comboBox_9.addItem("")
        self.comboBox_9.addItem("")
        self.comboBox_9.addItem("")
        self.gridLayout_8.addWidget(self.comboBox_9, 4, 4, 1, 1)
        self.comboBox_6 = QtWidgets.QComboBox(self.groupBox)
        self.comboBox_6.setObjectName("comboBox_6")
        self.comboBox_6.addItem("")
        self.comboBox_6.addItem("")
        self.comboBox_6.addItem("")
        self.comboBox_6.addItem("")
        self.gridLayout_8.addWidget(self.comboBox_6, 3, 4, 1, 1)
        self.comboBox_7 = QtWidgets.QComboBox(self.groupBox)
        self.comboBox_7.setObjectName("comboBox_7")
        self.comboBox_7.addItem("")
        self.comboBox_7.addItem("")
        self.comboBox_7.addItem("")
        self.comboBox_7.addItem("")
        self.gridLayout_8.addWidget(self.comboBox_7, 5, 4, 1, 1)
        self.label = QtWidgets.QLabel(self.groupBox)
        self.label.setObjectName("label")
        self.gridLayout_8.addWidget(self.label, 0, 4, 1, 1)
        self.gridLayout_9.addWidget(self.groupBox, 3, 0, 1, 1)
        self.FuzzingField = QtWidgets.QGroupBox(self.FieldandFuzzing)
        self.FuzzingField.setMinimumSize(QtCore.QSize(150, 0))
        self.FuzzingField.setObjectName("FuzzingField")
        self.SelectedPacketLabel_2 = QtWidgets.QLabel(self.FuzzingField)
        self.SelectedPacketLabel_2.setGeometry(QtCore.QRect(10, 50, 111, 31))
        self.SelectedPacketLabel_2.setObjectName("SelectedPacketLabel_2")
        self.SelectedPackeInput_2 = QtWidgets.QPlainTextEdit(self.FuzzingField)
        self.SelectedPackeInput_2.setGeometry(QtCore.QRect(140, 60, 141, 31))
        self.SelectedPackeInput_2.setObjectName("SelectedPackeInput_2")
        self.SelectedFieldName_2 = QtWidgets.QGroupBox(self.FuzzingField)
        self.SelectedFieldName_2.setGeometry(QtCore.QRect(10, 100, 271, 151))
        self.SelectedFieldName_2.setTitle("")
        self.SelectedFieldName_2.setObjectName("SelectedFieldName_2")
        self.SelectedField_2 = QtWidgets.QLabel(self.SelectedFieldName_2)
        self.SelectedField_2.setGeometry(QtCore.QRect(10, 10, 111, 16))
        self.SelectedField_2.setObjectName("SelectedField_2")
        self.ExpectedReturn_2 = QtWidgets.QLabel(self.SelectedFieldName_2)
        self.ExpectedReturn_2.setGeometry(QtCore.QRect(10, 40, 111, 16))
        self.ExpectedReturn_2.setObjectName("ExpectedReturn_2")
        self.Maximum_3 = QtWidgets.QLabel(self.SelectedFieldName_2)
        self.Maximum_3.setGeometry(QtCore.QRect(10, 100, 71, 16))
        self.Maximum_3.setObjectName("Maximum_3")
        self.Minimum_3 = QtWidgets.QLabel(self.SelectedFieldName_2)
        self.Minimum_3.setGeometry(QtCore.QRect(10, 70, 81, 16))
        self.Minimum_3.setObjectName("Minimum_3")
        self.Selected_2 = QtWidgets.QPlainTextEdit(self.SelectedFieldName_2)
        self.Selected_2.setGeometry(QtCore.QRect(130, 10, 141, 31))
        self.Selected_2.setObjectName("Selected_2")
        self.Expected_2 = QtWidgets.QPlainTextEdit(self.SelectedFieldName_2)
        self.Expected_2.setGeometry(QtCore.QRect(130, 40, 141, 31))
        self.Expected_2.setObjectName("Expected_2")
        self.Minimum_4 = QtWidgets.QPlainTextEdit(self.SelectedFieldName_2)
        self.Minimum_4.setGeometry(QtCore.QRect(130, 70, 141, 31))
        self.Minimum_4.setObjectName("Minimum_4")
        self.Maximum_4 = QtWidgets.QPlainTextEdit(self.SelectedFieldName_2)
        self.Maximum_4.setGeometry(QtCore.QRect(130, 100, 141, 31))
        self.Maximum_4.setObjectName("Maximum_4")
        self.Fuzz_2 = QtWidgets.QPushButton(self.FuzzingField)
        self.Fuzz_2.setGeometry(QtCore.QRect(100, 260, 75, 23))
        self.Fuzz_2.setObjectName("Fuzz_2")
        self.Stop_2 = QtWidgets.QPushButton(self.FuzzingField)
        self.Stop_2.setGeometry(QtCore.QRect(190, 260, 75, 23))
        self.Stop_2.setObjectName("Stop_2")
        self.gridLayout_9.addWidget(self.FuzzingField, 3, 3, 1, 1)
        self.gridLayout_3 = QtWidgets.QGridLayout()
        self.gridLayout_3.setObjectName("gridLayout_3")
        self.gridLayout_9.addLayout(self.gridLayout_3, 0, 0, 1, 1)
        self.Arrows = QtWidgets.QGroupBox(self.FieldandFuzzing)
        self.Arrows.setMaximumSize(QtCore.QSize(30, 100))
        self.Arrows.setTitle("")
        self.Arrows.setObjectName("Arrows")
        self.pushButton = QtWidgets.QPushButton(self.Arrows)
        self.pushButton.setGeometry(QtCore.QRect(0, 0, 31, 31))
        self.pushButton.setObjectName("pushButton")
        self.pushButton_2 = QtWidgets.QPushButton(self.Arrows)
        self.pushButton_2.setGeometry(QtCore.QRect(0, 70, 31, 31))
        self.pushButton_2.setObjectName("pushButton_2")
        self.gridLayout_9.addWidget(self.Arrows, 3, 1, 1, 1)
        self.gridLayout.addWidget(self.FieldandFuzzing, 3, 0, 1, 1)
        self.horizontalLayout_2.addLayout(self.gridLayout)

        self.retranslateUi(Form)
        self.PacketView_2.setCurrentIndex(0)
        QtCore.QMetaObject.connectSlotsByName(Form)

    def retranslateUi(self, Form):
        _translate = QtCore.QCoreApplication.translate
        Form.setWindowTitle(_translate("Form", "Form"))
        self.PcapFile.setTitle(_translate("Form", "PCAP File"))
        self.PCAFile_2.setText(_translate("Form", "PCAP File"))
        self.PCAPLocation_2.setPlainText(_translate("Form", "PCAP File Path"))
        self.OpenButton_2.setText(_translate("Form", "Open"))
        self.CancelButton_2.setText(_translate("Form", "Cancel"))
        self.PacketArea.setTitle(_translate("Form", "Packet Area"))
        self.ClearButton_4.setText(_translate("Form", "Clear"))
        __sortingEnabled = self.treeWidget.isSortingEnabled()
        self.treeWidget.setSortingEnabled(False)
        self.treeWidget.topLevelItem(0).setText(0, _translate("Form", "Frame 222"))
        self.treeWidget.topLevelItem(0).child(0).setText(0, _translate("Form", "IP"))
        self.treeWidget.topLevelItem(0).child(1).setText(0, _translate("Form", "DNS"))
        self.treeWidget.topLevelItem(1).setText(0, _translate("Form", "Frame 223"))
        self.treeWidget.topLevelItem(1).child(0).setText(0, _translate("Form", "IP"))
<<<<<<< HEAD
        #self.childItems = []
        #for i in range(2):
        #    self.childItems.append(QtWidgets.QTreeWidgetItem())
        #    self.treeWidget.topLevelItem(1).addChild(self.childItems[i])
        #    self.treeWidget.setItemWidget(self.childItems[0], 0, QtWidgets.QTreeWidgetItem())
        for i in range(2,4):
            self.treeWidget.topLevelItem(i).setText(0, _translate("Form", "Kaboob"))
            self.treeWidget.topLevelItem(i).child(1).setText(0, _translate("Form", "Boob"))
            self.treeWidget.topLevelItem(i).child(0).setText(0, _translate("Form", "Francisco Likes Anime Boobs"))

            #self.treeWidget.topLevelItem(i).child(0).setFlags(self.treeWidget.topLevelItem(0).child(0).flags() | QtCore.Qt.ItemIsEditable)
       # self.treeWidget.topLevelItem(2).child(1).setText(0, _translate("Form", "BO0OB"))
=======
>>>>>>> 4bc1659acff2391f05aad74e7d466a2641c221f2
        self.treeWidget.setSortingEnabled(__sortingEnabled)
        self.PacketView_2.setTabText(self.PacketView_2.indexOf(self.Dissected_2), _translate("Form", "Dissected"))
        __sortingEnabled = self.treeWidget_2.isSortingEnabled()
        self.treeWidget_2.setSortingEnabled(False)
        self.treeWidget_2.topLevelItem(0).setText(0, _translate("Form", "Frame 222"))
        self.treeWidget_2.topLevelItem(0).child(0).setText(0, _translate("Form", "Bin Dump"))
        self.treeWidget_2.topLevelItem(0).child(1).setText(0, _translate("Form", "Bin Dump"))
        self.treeWidget_2.topLevelItem(1).setText(0, _translate("Form", "Frame 223"))
        self.treeWidget_2.topLevelItem(1).child(0).setText(0, _translate("Form", "Bin Dump"))
        self.treeWidget_2.setSortingEnabled(__sortingEnabled)
        self.PacketView_2.setTabText(self.PacketView_2.indexOf(self.Binary_2), _translate("Form", "Binary"))
        __sortingEnabled = self.treeWidget_3.isSortingEnabled()
        self.treeWidget_3.setSortingEnabled(False)
        self.treeWidget_3.topLevelItem(0).setText(0, _translate("Form", "Frame 222"))
        self.treeWidget_3.topLevelItem(0).child(0).setText(0, _translate("Form", "Hex Dump"))
        self.treeWidget_3.topLevelItem(0).child(1).setText(0, _translate("Form", "Hex Dump"))
        self.treeWidget_3.topLevelItem(1).setText(0, _translate("Form", "Frame 223"))
        self.treeWidget_3.topLevelItem(1).child(0).setText(0, _translate("Form", "Hex Dump"))
        self.treeWidget_3.setSortingEnabled(__sortingEnabled)
        self.PacketView_2.setTabText(self.PacketView_2.indexOf(self.HEX_2), _translate("Form", "HEX"))
        self.CapturFilterArea.setTitle(_translate("Form", "Capture Filter"))
        self.FilterLabel_2.setText(_translate("Form", "Filter"))
        self.FilterExpressionInput_2.setPlainText(_translate("Form", "Filter Expression"))
        self.ApplyButton_2.setText(_translate("Form", "Apply"))
        self.ClearButton_3.setText(_translate("Form", "Clear"))
        self.groupBox.setTitle(_translate("Form", "Field Area"))
        self.Forward_2.setText(_translate("Form", "Forward"))
        self.SaveMod_2.setText(_translate("Form", "Save Modification"))
        self.Drop_2.setText(_translate("Form", "Drop"))
        self.comboBox_8.setItemText(0, _translate("Form", "Format"))
        self.comboBox_8.setItemText(1, _translate("Form", "Dissected"))
        self.comboBox_8.setItemText(2, _translate("Form", "Hex"))
        self.comboBox_8.setItemText(3, _translate("Form", "Binary"))
        self.FieldValuesCol_2.headerItem().setText(0, _translate("Form", "Field Name"))
        self.FieldValuesCol_2.headerItem().setText(1, _translate("Form", "Value"))
        self.FieldValuesCol_2.headerItem().setText(2, _translate("Form", "Mask"))
        __sortingEnabled = self.FieldValuesCol_2.isSortingEnabled()
        self.FieldValuesCol_2.setSortingEnabled(False)
        self.FieldValuesCol_2.topLevelItem(0).setText(0, _translate("Form", "icmp.type"))
        self.FieldValuesCol_2.topLevelItem(0).setText(1, _translate("Form", "08"))
        self.FieldValuesCol_2.topLevelItem(0).setText(2, _translate("Form", "0"))
        self.FieldValuesCol_2.topLevelItem(1).setText(0, _translate("Form", "icmp.code"))
        self.FieldValuesCol_2.topLevelItem(1).setText(1, _translate("Form", "00"))
        self.FieldValuesCol_2.topLevelItem(1).setText(2, _translate("Form", "0"))
        self.FieldValuesCol_2.topLevelItem(2).setText(0, _translate("Form", "icmp.checksum"))
        self.FieldValuesCol_2.topLevelItem(2).setText(1, _translate("Form", "6861"))
        self.FieldValuesCol_2.topLevelItem(2).setText(2, _translate("Form", ""))
        self.FieldValuesCol_2.topLevelItem(3).setText(0, _translate("Form", "icmp.ident"))
        self.FieldValuesCol_2.topLevelItem(3).setText(1, _translate("Form", "809e"))
        self.FieldValuesCol_2.topLevelItem(3).setText(2, _translate("Form", "0"))
        self.FieldValuesCol_2.topLevelItem(4).setText(0, _translate("Form", "icmp.seq"))
        self.FieldValuesCol_2.topLevelItem(4).setText(1, _translate("Form", "0f00"))
        self.FieldValuesCol_2.topLevelItem(4).setText(2, _translate("Form", "2"))
        self.FieldValuesCol_2.setSortingEnabled(__sortingEnabled)
        self.label_2.setText(_translate("Form", "Field Nmae, value, and display format are editable fields"))
        self.comboBox_10.setItemText(0, _translate("Form", "Format"))
        self.comboBox_10.setItemText(1, _translate("Form", "Dissected"))
        self.comboBox_10.setItemText(2, _translate("Form", "Hex"))
        self.comboBox_10.setItemText(3, _translate("Form", "Binary"))
        self.comboBox_9.setItemText(0, _translate("Form", "Format"))
        self.comboBox_9.setItemText(1, _translate("Form", "Dissected"))
        self.comboBox_9.setItemText(2, _translate("Form", "Hex"))
        self.comboBox_9.setItemText(3, _translate("Form", "Binary"))
        self.comboBox_6.setItemText(0, _translate("Form", "Format"))
        self.comboBox_6.setItemText(1, _translate("Form", "Dissected"))
        self.comboBox_6.setItemText(2, _translate("Form", "Hex"))
        self.comboBox_6.setItemText(3, _translate("Form", "Binary"))
        self.comboBox_7.setItemText(0, _translate("Form", "Format"))
        self.comboBox_7.setItemText(1, _translate("Form", "Dissected"))
        self.comboBox_7.setItemText(2, _translate("Form", "Hex"))
        self.comboBox_7.setItemText(3, _translate("Form", "Binary"))
        self.label.setText(_translate("Form", "Display Format"))
        self.FuzzingField.setTitle(_translate("Form", "Fuzzing Area"))
        self.SelectedPacketLabel_2.setText(_translate("Form", "Selected Packet Name"))
        self.SelectedPackeInput_2.setPlainText(_translate("Form", "Selected Packet Name"))
        self.SelectedField_2.setText(_translate("Form", "Selected Field Name"))
        self.ExpectedReturn_2.setText(_translate("Form", "Expected Return Type"))
        self.Maximum_3.setText(_translate("Form", "Maximum"))
        self.Minimum_3.setText(_translate("Form", "Minimum"))
        self.Selected_2.setPlainText(_translate("Form", "Selected Field Name"))
        self.Expected_2.setPlainText(_translate("Form", "Expected Return Type"))
        self.Minimum_4.setPlainText(_translate("Form", "Minimum"))
        self.Maximum_4.setPlainText(_translate("Form", "Maximum"))
        self.Fuzz_2.setText(_translate("Form", "Fuzz"))
        self.Stop_2.setText(_translate("Form", "Stop"))
        self.pushButton.setText(_translate("Form", "|+>"))
        self.pushButton_2.setText(_translate("Form", "|+>"))
<<<<<<< HEAD
       # for i in range(7):
 


    def addDropDownMumboJumbo(self,parent):
        return QtWidgets.QTreeWidgetItem(parent)

    def openPCAP(self):
        pcap = self.PCAPLocation_2.toPlainText()
        packets, pktsData = scapyUtilities.unpackPCAP(pcap)
        print('Imported PCAP')

        #trees = Tree(self.Dissected_2, pktsData, packets)

    def updatePCAPArea(self):
        _translate = QtCore.QCoreApplication.translate
        self.treeWidget.topLevelItem(2).setText(0, _translate("Form", "Kaboob"))
        self.treeWidget.topLevelItem(2).child(0).setText(0, _translate("Form", "Boob"))
        self.treeWidget.topLevelItem(2).child(0).setFlags(self.treeWidget.topLevelItem(0).child(0).flags() | QtCore.Qt.ItemIsEditable)
        self.treeWidget.topLevelItem(2).child(1).setText(0, _translate("Form", "BO0OB"))

"""
class Tree(QtWidgets.QTreeWidget):
    def __init__(self, treeWidget, pktsData, packets):
        super(Tree, self).__init__()
        self.pktsData = pktsData
        self.treeWidget = treeWidget
        for i, packet in enumerate(self.pktsData):

            rootItem = QtWidgets.QTreeWidgetItem()
            rootItem.setFlags(rootItem.flags() | QtCore.Qt.ItemIsEditable)
            l=list(scapyUtilities.dissectLayers(packets[i]))
            rootItem.setText(0, 'frame %s: %s'%(i, str(l)))
            print('frame %s: %s'%(i, str(l)))
            rootItem.sceneSG={}
            rootItem.sceneSG['code']='nextSceneFilename'

            for layer in packet:
                print(type(layer))
                for layername, layerInfo in layer.items():
                    childItem = QtWidgets.QTreeWidgetItem(rootItem)
                    childItem.setFlags(rootItem.flags() | QtCore.Qt.ItemIsEditable)
                    childItem.sceneSG={}     
                    childItem.setText(0, '%s: %s' %(layername, str(layerInfo)))

            rootItem.setData(100, 77, QtCore.Qt.UserRole )

            #self.addTopLevelItem(rootItem)

        #self.setColumnCount(1)
        self.doubleClicked.connect(self.onDoubleClick)
       # self.show()

    def onDoubleClick(self, index):
        item = self.currentItem()
        item.setFlags(item.flags() | QtCore.Qt.ItemIsEditable)
        if index.column()!=0:
            item.setFlags(item.flags() & ~QtCore.Qt.ItemIsEditable)
"""
=======

>>>>>>> 4bc1659acff2391f05aad74e7d466a2641c221f2
